src/peep/peep.c

```diff
@@ -7772,13 +7772,404 @@ static void peep_on_enter_or_exit_ride(rct_peep *peep, int rideIndex, int flags)
+	}
+
+loc_69B221:
+	uint32 edi = rideIndex + 0x0260; // This is used way at the end but is assigned here.
+	if (!RCT2_GLOBAL(0x00F1AEE3, unsigned char)) { // If no voucher
+		if (shopItem >= 32) {
+			// This seems to be identical to the construct above.
+			dx = RCT2_ADDRESS(0x00982246, uint16)[shopItem2 * 4];
+			if (RCT2_ADDRESS_CURRENT_TEMPERATURE >= 21)
+				dx = RCT2_ADDRESS(0x00982248, uint16)[shopItem2 * 4];
+			if (RCT2_ADDRESS_CURRENT_TEMPERATURE <= 11)
+				dx = RCT2_ADDRESS(0x0098224A, uint16)[shopItem2 * 4];
+		}
+		else {
+			dx = RCT2_ADDRESS(0x00982166, uint16)[shopItem2 * 4];
+			if (RCT2_ADDRESS_CURRENT_TEMPERATURE >= 21)
+				dx = RCT2_ADDRESS(0x00982168, uint16)[shopItem2 * 4];
+			if (RCT2_ADDRESS_CURRENT_TEMPERATURE <= 11)
+				dx = RCT2_ADDRESS(0x0098216A, uint16)[shopItem2 * 4];
+		}
+		/// loc_69B293:
+		dx -= price;
+		uint8 bl = 0; // Satisfaction incremental
+		if (dx > 0xFFF8) {
+			bl++;
+			if (dx > 0xFFFD) {
+				bl++;
+				if (dx > 3)
+					bl++;
+			}
+		}
+		/// loc_69B2B0:
+		// FIXME: need to make sure the ride pointer is passed in
+		// to be able to call this correctly.
+		// May require different params to be made available from higher up.
+		ride_update_satisfaction(ride, bl);
+	}
+///loc_69B2B7:
+	// The peep has now decided to buy the item (or, specifically, has not
+	// been dissuaded so far).
+	// FIXME: Is there a call to have the peep buy an item?
+	if (shopItem >= 32)
+		peep->item_extra_flags |= (1 << shopItem2);
+	else
+		peep->item_standard_flags |= (1 << shopItem2);
+
+/// loc_69B2C8:
+	if (shopItem == 20) // T-Shirt
+		peep->tshirt_colour = RCT2_ADDRESS(0x1362AB4, uint8)[rideIndex * 0x260];
+
+/// loc_69B2E0:
+	if (shopItem == 18) // Hat
+		peep->hat_colour = RCT2_ADDRESS(0x1362AB4, uint8)[rideIndex * 0x260];
+
+/// loc_69B2FB:
+	if (shopItem == 0) // Balloon
+		peep->balloon_colour = RCT2_ADDRESS(0x1362AB4, uint8)[rideIndex * 0x260];
+
+/// loc_69B316:
+	if (shopItem == 4) // Umbrella
+		peep->umbrella_colour = RCT2_ADDRESS(0x1362AB4, uint8)[rideIndex * 0x260];
+
+/// loc_69B331:
+	if (shopItem == 2) // Map
+		sub_69A98C(peep);
+
+/// loc_69B33A:
+	uint16 dl;
+	if (shopItem >= 32)
+		dl = RCT2_ADDRESS(0x982310, uint8)[shopItem2];
+	else
+		dl = RCT2_ADDRESS(0x9822F4, uint8)[shopItem2];
+/// loc_69B34C:
+	peep->var_42 = min((peep->var_42 + dl), 255);
+
+/// loc_69B355:
+	if (shopItem == 3)
+		peep->photo1_ride_ref = rideIndex;
+
+/// loc_69B35F:
+	if (shopItem == 32)
+		peep->photo2_ride_ref = rideIndex;
+
+/// loc_69B366:
+	if (shopItem == 33)
+		peep->photo3_ride_ref = rideIndex;
+
+/// loc_69B36D:
+	if (shopItem == 34)
+		peep->photo4_ride_ref = rideIndex;
+
+/// loc_69B374:
+	peep->window_invalidate_flags |= PEEP_INVALIDATE_PEEP_INVENTORY;
+	peep_update_sprite_type(peep);
+	uint32 ecx = price;
+	uint32 ebx;
+	if (peep->flags & PEEP_FLAGS_TRACKING) {
+		RCT2_GLOBAL(0x13CE952,uint16) = peep->name_string_idx;
+		RCT2_GLOBAL((0x13CE952 + 2), uint32) = peep->id;
+		uint16 bx = shopItem + 0x7F;
```

Comment:
This is an `rct_string_id` try to remember to add entries into the string_ids.h file.